types/parse/index.d.ts

```diff
@@ -272,31 +278,32 @@ declare namespace Parse {
      * @param {Object} attributes The initial set of data to store in the object.
      * @param {Object} options The options for this object instance.
      * @see Parse.Object.extend
+     * @typedef K   represents the interface for attributes
      *
      * @class
      *
      * Creates a new model with defined attributes.
      */
-    class Object extends BaseObject {
+    class Object<K = any> extends BaseObject {
 
         id: string;
         createdAt: Date;
         updatedAt: Date;
-        attributes: any;
+        attributes: K;
         cid: string;
         changed: boolean;
         className: string;
 
         constructor(className?: string, options?: any);
         constructor(attributes?: string[], options?: any);
 
-        static createWithoutData<T extends Object>(id: string): T;
-        static destroyAll<T extends Object>(list: T[], options?: Object.DestroyAllOptions): Promise<T[]>;
+        static createWithoutData<T extends Object>(this: { new(): T }, id: string): T;
+        static destroyAll<T extends Object>(this: { new(): T }, list: T[], options?: Object.DestroyAllOptions): Promise<T[]>;
         static extend(className: string | { className: string }, protoProps?: any, classProps?: any): any;
         static fetchAll<T extends Object>(list: T[], options: Object.FetchAllOptions): Promise<T[]>;
         static fetchAllIfNeeded<T extends Object>(list: T[], options: Object.FetchAllOptions): Promise<T[]>;
         static fetchAllWithInclude<T extends Object>(list: T[], keys: string | Array<string | Array<string>>, options: RequestOptions): Promise<T[]>;
-        static fromJSON(json: any, override?: boolean): Object;
+        static fromJSON<T>(this: { new(): T }, json: any, override?: boolean): T;
```

Comment:
What about `json`'s type as the `K` for the returned `Object<K>`?

```suggestion
        static fromJSON<T extends Object<K>, K>(this: { new(): T }, json: K, override?: boolean): T;
```

Reply:
It would indeed make sense to add those as well. 
Unfortunately I'm struggling with the User type:

Adding the generic to User and passing it along to Object causes `Query` to infer the type incorrectly. 
```typescript
class User<K = any> extends Object<K> { }
class Query<T extends Object, K = Unobject<T>> extends BaseObject { }
```
causes the following to fail:
```typescript
const queryB = new Parse.Query(Parse.User);
queryB.near("a", userGeoPoint); // Argument of type '"a"' is not assignable to parameter of type 'never'.
```
The type for `queryB` was set to `const queryB: Parse.Query<Parse.User<{}>, {}>`, where `Parse.Query<Parse.User<any>, any>` was expected.

Reply:
@dirkgroenen, sorry for the long time without replying.

The issue you pointed out is also a problem for `Parse.Object`. The problem is in the inference of `T` by the type of the constructor's parameter `objectClass`. If you do `const queryB = new Parse.Query(Parse.Object);`, you will see that the `typeof queryB` is also wrong (`Parse.Query<Parse.Object<unknown>, unknown>`).

I am not sure why, but if you do `const queryB = new Parse.Query(typeof ({} as Parse.Object));` it works. Another option that works is `const queryB = new Parse.Query<Parse.Object>(Parse.Object);`. However, a possible solution for this would be changing the constructor of `Parse.Query`. I am going to send you a suggestion for that in a new thread.

But what about the suggested change I proposed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/36172#discussion_r294058705?

Reply:
The `fromJson` suggestion doesn't seem to work correctly. Using `Unobject` does give us the desired result:
```
static fromJSON<T extends Object<K>, K>(this: { new(): T }, json: K, override?: boolean): T;
```
![image](https://user-images.githubusercontent.com/1001391/61212814-93ec7b00-a703-11e9-9c25-514761862433.png)